### ISSUE TITLE
Extension not found when parent folder is same name as extension directory

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-cli#2351]
Let's say I have this structure: 
````
> tree
C:\USERS\CHRIS\APPDATA\LOCAL\TEMP\TEST-LIGHTBOX
└───lightbox
    ├───example_files
    │   └───mediabag
    ├───images
    └───_extensions
        └───quarto-ext
            └───lightbox
                └───resources
                    ├───css
                    └───js
````

And I am place in the root `test-lightbox` repo 
````
> pwd

Path
----
C:\Users\chris\AppData\Local\Temp\test-lightbox
````

Should this work as `_extensions` is correctly at the level of the document ? 

````
> quarto render .\lightbox\example.qmd --to html
pandoc
  to: html
  output-file: example.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Example Lightbox Document
  lightbox: auto

Error running filter lightbox:
Could not find executable lightbox
````

If I enter the folder of the `.qmd` then it works. 

````
> cd .\lightbox\
> quarto render example.qmd --to html
pandoc
  to: html
  output-file: example.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png

metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  title: Example Lightbox Document
  lightbox: auto

Output created: example.html
````

Trying to check we don't have a regression here.

_Originally posted by @cderv in https://github.com/quarto-dev/quarto-cli/discussions/2331#discussioncomment-3586223_

Comment from @dragonstyle (https://github.com/quarto-dev/quarto-cli/discussions/2331#discussioncomment-3589355)

> Yeah this seems wrong (like we're using the working directory rather than the input directory when resolving the path to the extension). Extensions are a special case since there you are referencing an extension name rather than a filter path, but this should definitely work.

**Comment by cderv** [comment]
In fact @dragonstyle the issue happens only if the subfolder is called the same name as the extension called . It works if I rename the folder to another name

````powershell
C:\USERS\CHRIS\APPDATA\LOCAL\TEMP\TEST-LIGHTBOX
└───sub
    ├───test_files
    │   └───libs
    │       ├───bootstrap
    │       ├───clipboard
    │       └───quarto-html
    └───_extensions
        └───quarto-ext
            └───lightbox
                └───resources
                    ├───css
                    └───js
````

**Comment by dragonstyle** [comment]
This should be fixed as 1.2.246 (767dd2aaecebc04eeeea56178e22e34b127c3f63)!
